### PR TITLE
Rework of build section of the wiki. Migrate, merge, and rename pages.

### PR DESCRIPTION
**Pull request by nuke-web3**
Major overhaul of the Builders section. Open to review and comments. This mostly is about consolidation of duplicated content and bringing pages together. It removes some outdated content as well. 

Another intention is to remove any and all content that references a time, and removes any mention of the "eventual launch" of Polkadot and other projects. 

Checks before a merge:
- [ ] Fact check on new content
- [ ] No broken links from page deletion/renaming
- [x] Spelling 🤞🏼 
- [x] Remove all references to specific time (this will be/do...)
- [ ] Modify Rococo references and/or add Westend details for parachain testing (likely a followup PR)

A few things to consider in follow up PRs:
- Remove all notion of parachain/thread cost comparisons.
- Remove all reference of _specific_ projects and teams (bias) , or add a living reference to something like polkaproject (but scam concerns taken into account) 
- Use "relay chain" when not explicitly referring to Polkadot or Kusama (as a proper noun). This is done for parachains already.

### CODE REVIEW

**Comment by Swader** [comment]
Good direction. I think @joepetrowski will be interested in taking a look at this too since he did most of the work on the integration / building guide stuff.

File: docs/build-parachains.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+---
+id: build-parachains
+title: Parachain Development
```
Comment (joepetrowski) [inline docs/build-parachains.md:3]: Why didn't you use `git mv`?

File: docs/build-index.md [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
-  be deployed to a live chain.
+- [Polkadot Smart Contracts](build-smart-contracts) - Tools to deploy contracts to a Polkadot parachain.
 
 ## Integration Guide
 
 This section is for service providers like wallets or node operators who want to learn how to
 integrate Polkadot into their systems.
 
-- [Integration Initiation](build-integration) - The guide to the guide.
+- [Integration Initiation](build-integration) - An index of tools and resources.
```
Comment (joepetrowski) [inline docs/build-index.md]:
```suggestion
- [Integration Initiation](build-integration) - The guide to the guide.
```

**Comment by joepetrowski** [comment]
Don't have time to review everything but you need to use `git mv` and not just copy/paste files (killing the git history).

File: docs/build-parachains.md (hunk 1 above)
Comment (nuke-web3) [reply docs/build-parachains.md:3]: Never needed it before TBH. It's usually picked up by git automagically. I didn't use this for the `build-build-with-polkadot.md` -> `build-guide.md` in this PR, as an example 🤷🏼 
TIL

**Comment by nuke-web3** [comment]
Sorry for the issue here, Joe. Closing in favor of #2076

**Pull request closed without merging**